libcaf_net/src/net/web_socket/framing.cpp

```diff
@@ -60,24 +61,27 @@ ptrdiff_t framing::consume(byte_span buffer, byte_span) {
     if (opcode_ == nil_code) {
       // Call upper layer.
       if (hdr.opcode == detail::rfc6455::connection_close) {
-        up_->abort(make_error(sec::connection_closed));
+        // TODO
+        abort_and_close_connection(sec::connection_closed, "");
         return -1;
       } else if (!handle(hdr.opcode, payload)) {
         return -1;
       }
     } else if (hdr.opcode != detail::rfc6455::continuation_frame) {
       CAF_LOG_DEBUG("expected a WebSocket continuation_frame");
-      up_->abort(make_error(sec::protocol_error,
-                            "expected a WebSocket continuation_frame"));
+      abort_and_close_connection(sec::protocol_error,
+                                 "expected a WebSocket continuation_frame");
       return -1;
     } else if (payload_buf_.size() + payload_len > max_frame_size) {
       CAF_LOG_DEBUG("fragmented WebSocket payload exceeds maximum size");
-      up_->abort(make_error(sec::protocol_error, "fragmented WebSocket payload "
-                                                 "exceeds maximum size"));
+      abort_and_close_connection(sec::protocol_error,
+                                 "fragmented WebSocket payload "
+                                 "exceeds maximum size");
       return -1;
     } else {
       if (hdr.opcode == detail::rfc6455::connection_close) {
-        up_->abort(make_error(sec::connection_closed));
+        // TODO
```

Comment:
What's the todo item?

Reply:
Fixed the question I've had with variadic template.